src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,49 @@
+
 package core.stack;
 
+import java.util.Arrays;
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_SIZE = 8;
+    private Object[] stack;
+    private int size = 0;
 
+    public MateStack() {
+        stack = new Object[DEFAULT_SIZE];
+        Arrays.fill(stack, null);
+    }
+
+    public void push(T value) {
+        if (size < stack.length) {
```

Comment:
The stack implementation should not have a fixed size. Instead of throwing StackOverflowError when the array is full, you should create a larger array and copy the old elements into it.